src/controllers/v1/registration.js

```diff
@@ -186,13 +186,31 @@ const RegistrationController = {
    */
   delete: async (id, cleanObject) => {
     try {
+      // First find all attached returns.
+      const returns = await Return.findAll({where: {RegistrationId: id}});
+      // Create an array of the return ids.
+      const returnIds = returns.map((meatBaitReturn) => meatBaitReturn.id);
+      // Start the transaction.
       await db.sequelize.transaction(async (t) => {
+        // Check the registration exists.
         await Registration.findByPk(id, {transaction: t, rejectOnEmpty: true});
+        // Create the revocation entry.
         await Revocation.create(cleanObject, {transaction: t});
+        // If there was any returns found we need to delete them also.
+        if (returns) {
+          // Soft Delete any nonTargetSpecies attached to returns.
+          await NonTargetSpecies.destroy({where: {ReturnId: returnIds}, transaction: t});
```

Comment:
Could you add a comment here to explain that you're going two layers deep for your deletes? Reg -> Return -> NTSpecies? I looked at this and the SFO version side by side and it took a while to work out why you were `findAll`ing the returns and the `id`s up the top and using this `if` here until I worked out that you needed to go two layers down.